app/models/minion.rb

```diff
@@ -15,20 +15,20 @@ class NonExistingMinion < StandardError; end
   # Example:
   #   Minion.assign_roles(
   #     roles: {
-  #       "master.example.com" => ["master"],
-  #       "minion1.example.com" => ["minion"]
+  #       master: 1,
+  #       minion: [2, 3]
   #     },
   #     default_role: :dns
   #   )
   def self.assign_roles!(roles: {}, default_role: :minion)
-    requested_master = roles.detect { |_name, r| r.include?("master") }.first
-    requested_minions = roles.select { |_name, r| r.include?("minion") }.keys
-    if roles.values.flatten.include?("master") && !Minion.exists?(hostname: requested_master)
-      raise NonExistingMinion, "Failed to process non existing minion: #{requested_master}"
+    requested_master = roles[:master]
+    requested_minions = roles[:minion]
+    if roles[:master] && !Minion.exists?(id: requested_master)
```

Comment:
I'd use `!requested_master.blank? && !Minion.exists?(id: requested_master)`. In this kind of checks when taking external information is better to use `blank?` because it also checks for emptyness, also refer to `requested_master` since we already bound the variable.